### ISSUE TITLE
[BUG]Attempt to invoke virtual method 'int java.lang.Integer.intValue()' on a null object reference in 1.0.2

### ISSUE THREAD

**Issue by dumptyhumpty2014gmail** [transistorsoft/flutter_background_fetch#255]
**Your Environment**
* Plugin version:
 1.0.2
* Platform: iOS or Android
Android
* OS version:
10
* Device manufacturer / model:
* Flutter info (`flutter info`, `flutter doctor`):
* [√] Flutter (Channel stable, 2.5.3, on Microsoft Windows [Version 10.0.19043.1348], locale ru-RU)
[√] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
[√] Chrome - develop for the web
[√] Android Studio (version 2020.3)
[!] Android Studio (version 2021.1)
    X Unable to determine bundled Java version.
[√] VS Code (version 1.62.2)
[√] Connected device (3 available)

* Plugin config

**To Reproduce**
1. Run example
2.
3.
4.

**Debug logs**
- ios XCode logs, 
- Android: `$ adb logcat`
I/flutter ( 7074): [BackgroundFetch] configure success: 2
E/MethodChannel#com.transistorsoft/flutter_background_fetch/methods( 7074): Failed to handle method call
E/MethodChannel#com.transistorsoft/flutter_background_fetch/methods( 7074): java.lang.NullPointerException: Attempt to invoke virtual method 'int java.lang.Integer.intValue()' on a null object reference
E/MethodChannel#com.transistorsoft/flutter_background_fetch/methods( 7074): 	at com.transistorsoft.flutter.backgroundfetch.BackgroundFetchModule.buildConfig(BackgroundFetchModule.java:186)
E/MethodChannel#com.transistorsoft/flutter_background_fetch/methods( 7074): 	at com.transistorsoft.flutter.backgroundfetch.BackgroundFetchModule.scheduleTask(BackgroundFetchModule.java:156)
E/MethodChannel#com.transistorsoft/flutter_background_fetch/methods( 7074): 	at com.transistorsoft.flutter.backgroundfetch.BackgroundFetchModule.onMethodCall(BackgroundFetchModule.java:105)
E/MethodChannel#com.transistorsoft/flutter_background_fetch/methods( 7074): 	at io.flutter.plugin.common.MethodChannel$IncomingMethodCallHandler.onMessage(MethodChannel.java:233)
E/MethodChannel#com.transistorsoft/flutter_background_fetch/methods( 7074): 	at io.flutter.embedding.engine.dart.DartMessenger.handleMessageFromDart(DartMessenger.java:84)
E/MethodChannel#com.transistorsoft/flutter_background_fetch/methods( 7074): 	at io.flutter.embedding.engine.FlutterJNI.handlePlatformMessage(FlutterJNI.java:865)
E/MethodChannel#com.transistorsoft/flutter_background_fetch/methods( 7074): 	at android.os.MessageQueue.nativePollOnce(Native Method)
E/MethodChannel#com.transistorsoft/flutter_background_fetch/methods( 7074): 	at android.os.MessageQueue.next(MessageQueue.java:336)
E/MethodChannel#com.transistorsoft/flutter_background_fetch/methods( 7074): 	at android.os.Looper.loop(Looper.java:174)
E/MethodChannel#com.transistorsoft/flutter_background_fetch/methods( 7074): 	at android.app.ActivityThread.main(ActivityThread.java:7356)
E/MethodChannel#com.transistorsoft/flutter_background_fetch/methods( 7074): 	at java.lang.reflect.Method.invoke(Native Method)
E/MethodChannel#com.transistorsoft/flutter_background_fetch/methods( 7074): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
E/MethodChannel#com.transistorsoft/flutter_background_fetch/methods( 7074): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:930)
E/flutter ( 7074): [ERROR:flutter/lib/ui/ui_dart_state.cc(209)] Unhandled Exception: PlatformException(error, Attempt to invoke virtual method 'int java.lang.Integer.intValue()' on a null object reference, null, java.lang.NullPointerException: Attempt to invoke virtual method 'int java.lang.Integer.intValue()' on a null object reference
**Additional context**
In version 1.0.1 is OK.

**Comment by dumptyhumpty2014gmail** [comment]
In my app wich worked on 1.0.1 version I have this bug too. Now I fix 1.0.1 in pubscep.yaml but I'm interested in what needs to be done

**Comment by ahoelzemann** [comment]
I had similar issue and the reason why this did happen was that I didn't set the "requiredNetworkType" parameter to NetworkType.NONE.

this definition works:
```
await BackgroundFetch.configure(
        BackgroundFetchConfig(
            minimumFetchInterval: 60,
            stopOnTerminate: true,
            enableHeadless: true,
            requiresBatteryNotLow: false,
            requiresCharging: false,
            requiresStorageNotLow: false,
            requiresDeviceIdle: false,
            requiredNetworkType: NetworkType.NONE
        )
```
this definition doesn't work: 
```
await BackgroundFetch.configure(
        BackgroundFetchConfig(
            minimumFetchInterval: 60,
            stopOnTerminate: true,
            enableHeadless: true,
            requiresBatteryNotLow: false,
            requiresCharging: false,
            requiresStorageNotLow: false,
            requiresDeviceIdle: false,
   //         requiredNetworkType: NetworkType.NONE
        )
```
So I guess in the new version you need to explicitly set all the parameters.

**Comment by dumptyhumpty2014gmail** [comment]
My parameters:
      await BackgroundFetch.configure(
        BackgroundFetchConfig(
          minimumFetchInterval: 15,
          forceAlarmManager: false,
          stopOnTerminate: false,
          startOnBoot: false,
          enableHeadless: true,
          requiresBatteryNotLow: false,
          requiresCharging: false,
          requiresStorageNotLow: false,
          requiresDeviceIdle: false,
          requiredNetworkType: NetworkType.NONE,
        ),
and parameters for task:
  BackgroundFetch.scheduleTask(TaskConfig(
    taskId: taskId,
    delay: taskDelay,
    periodic: false,
    forceAlarmManager: true,
    stopOnTerminate: false,
    enableHeadless: true,
    requiresNetworkConnectivity:
        false, 
    requiresCharging: false,
  ));
In any case, thx. I will try to change parameters

**Comment by Acventor** [comment]
I've faced the similar problem. The thing is, I set the parameter `requiredNetworkType: NetworkType.ANY` in the `BackgroundFetch.configure`, but forgot to do the same for the `TaskConfig`. So after I added it to both of them, the problem disappeared.

**Comment by christocracy** [comment]
Must be something introduced in the plug-in's Dart api (which is very tiny and simple) related to null safety protection which has backfired.

**Comment by Bes79** [comment]
Same issue

**Comment by christocracy** [comment]
It will be the same solution as above.

**Comment by dumptyhumpty2014gmail** [comment]
Thx, "requiredNetworkType: NetworkType.ANY," in configure and in task solved

**Comment by christocracy** [comment]
The problem was a simple [typo with a copy/paste](https://github.com/transistorsoft/flutter_background_fetch/blob/master/lib/background_fetch.dart#L160-L162)

It's supposed to be like all the of `if (xxx != null)` blocks:

```dart
if (requiredNetworkType != null) {
      config['requiredNetworkType'] = requiredNetworkType?.index;
}  
```

**Comment by christocracy** [comment]
Fixed published to `1.0.3`.
